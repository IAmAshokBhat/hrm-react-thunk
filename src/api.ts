import {
  ILoginData,
  INewAttendance,
  INewLeave,
  IRegisterAPI
} from './constants';
import { fetchWithToken, getToken } from './utils';

const headers = {
  'X-JWT-Token': getToken()
};

export const getUserListAPI = () => fetchWithToken('user/list');

export const getUserDetailsAPI = (id: number) =>
  fetchWithToken(`api/users/${id}`);

export const getAttendanceDetailsAPI = (id: number, from: string, to: string) =>
  fetchWithToken(`api/attendance/${id}?from=${from}&to=${to}`);

export const getLeaveBalanceAPI = (id: number) =>
  fetchWithToken(`api/leaves/${id}`);

export const getEventDetailsAPI = (id: number, from: string, to: string) =>
  fetchWithToken(`api/events/${id}?from=${from}&to=${to}`);

export const getAllLeavesAPI = (id: number) =>
  fetchWithToken(`api/all-leaves/${id}`);

export const applyLeaveAPI = (newLeave: INewLeave) =>
  fetchWithToken(`api/apply-leave`, {
    method: 'POST',
    body: JSON.stringify(newLeave)
  });

export const addAttendanceAPI = (newAttendance: INewAttendance) =>
  fetchWithToken(`api/attendance`, {
    method: 'POST',
    body: JSON.stringify(newAttendance)
  });

export const loginAPI = (userInfo: ILoginData) =>
  fetchWithToken(`login`, {
    method: 'POST',
    body: JSON.stringify(userInfo)
  });

export const getDesignationListAPI = () =>
  fetchWithToken(`designation`, { headers: headers });

export const registerAPI = (newUser: IRegisterAPI) =>
  fetchWithToken(`user`, {
    method: 'POST',
    body: JSON.stringify(newUser)
  });
