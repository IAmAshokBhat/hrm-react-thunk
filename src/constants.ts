export interface ILeave {
  leaveType: string;
  from: Date;
  to: Date;
  status: string;
  noOfDays: number;
}

export interface ILeaveInfo {
  from: string;
  to: string;
  category: string;
}

export interface ISelectLeaveType {
  value: number;
  label: string;
  balance: number;
}

export enum ACTIONS {
  GET_USERS = 'GET_USERS',
  GET_USERS_LOADING = 'GET_USERS_LOADING',
  GET_USERS_SUCCESS = 'GET_USERS_SUCCESS',
  GET_USERS_FAILURE = 'GET_USERS_FAILURE',

  GET_USER_DETAILS = 'GET_USER_DETAILS',
  GET_USER_DETAILS_LOADING = 'GET_USER_DETAILS_LOADING',
  GET_USER_DETAILS_SUCCESS = 'GET_USER_DETAILS_SUCCESS',
  GET_USER_DETAILS_FAILURE = 'GET_USER_DETAILS_FAILURE',

  GET_ATTENDANCE_DETAILS = 'GET_ATTENDANCE_DETAILS',
  GET_ATTENDANCE_DETAILS_LOADING = 'GET_ATTENDANCE_DETAILS_LOADING',
  GET_ATTENDANCE_DETAILS_SUCCESS = 'GET_ATTENDANCE_DETAILS_SUCCESS',
  GET_ATTENDANCE_DETAILS_FAILURE = 'GET_ATTENDANCE_DETAILS_FAILURE',

  GET_LEAVE_BALANCE = 'GET_LEAVE_BALANCE',
  GET_LEAVE_BALANCE_LOADING = 'GET_LEAVE_BALANCE_LOADING',
  GET_LEAVE_BALANCE_SUCCESS = 'GET_LEAVE_BALANCE_SUCCESS',
  GET_LEAVE_BALANCE_FAILURE = 'GET_LEAVE_BALANCE_FAILURE',

  GET_EVENT_DETAILS = 'GET_EVENT_DETAILS',
  GET_EVENT_DETAILS_LOADING = 'GET_EVENT_DETAILS_LOADING',
  GET_EVENT_DETAILS_SUCCESS = 'GET_EVENT_DETAILS_SUCCESS',
  GET_EVENT_DETAILS_FAILURE = 'GET_EVENT_DETAILS_FAILURE',

  GET_ALL_LEAVES = 'GET_ALL_LEAVES',
  GET_ALL_LEAVES_LOADING = 'GET_ALL_LEAVES_LOADING',
  GET_ALL_LEAVES_SUCCESS = 'GET_ALL_LEAVES_SUCCESS',
  GET_ALL_LEAVES_FAILURE = 'GET_ALL_LEAVES_FAILURE',

  GET_APPLY_LEAVE = 'GET_APPLY_LEAVE',
  GET_APPLY_LEAVE_LOADING = 'GET_APPLY_LEAVE_LOADING',
  GET_APPLY_LEAVE_SUCCESS = 'GET_APPLY_LEAVE_SUCCESS',
  GET_APPLY_LEAVE_FAILURE = 'GET_APPLY_LEAVE_FAILURE',

  GET_ADD_ATTENDANCE = 'GET_ADD_ATTENDANCE',
  GET_ADD_ATTENDANCE_LOADING = 'GET_ADD_ATTENDANCE_LOADING',
  GET_ADD_ATTENDANCE_SUCCESS = 'GET_ADD_ATTENDANCE_SUCCESS',
  GET_ADD_ATTENDANCE_FAILURE = 'GET_ADD_ATTENDANCE_FAILURE',

  GET_LOGIN = 'GET_LOGIN',
  LOGIN_LOADING = 'GET_LOGIN_LOADING',
  LOGIN_SUCCESS = 'GET_LOGIN_SUCCESS',
  LOGIN_FAILURE = 'GET_LOGIN_FAILURE',

  GET_DESIGNATION_LIST = 'GET_DESIGNATION_LIST',
  GET_DESIGNATION_LIST_LOADING = 'GET_DESIGNATION_LIST_LOADING',
  GET_DESIGNATION_LIST_SUCCESS = 'GET_DESIGNATION_LIST_SUCCESS',
  GET_DESIGNATION_LIST_FAILURE = 'GET_DESIGNATION_LIST_FAILURE',

  REGISTER = 'REGISTER',
  REGISTER_LOADING = 'REGISTER_LOADING',
  REGISTER_SUCCESS = 'REGISTER_SUCCESS',
  REGISTER_FAILURE = 'REGISTER_FAILURE'
}

export interface IUserAPI {
  user_id: number;
  name: string;
}
export interface IUserListAPI {
  data: IUserAPI[];
}

export interface IUserInfoDataAPI {
  user_id: number;
  name: string;
  designation: string;
  manager: string;
  email: string;
}

export interface IUserInfoAPI {
  data: IUserInfoDataAPI[];
}

export interface IUserInfo {
  userId: number;
  name: string;
  designation: string;
  manager: string;
  email: string;
}

interface ILeaveBalanceItemAPI {
  leave_type_id: number;
  leave_type: string;
  balance: number;
}

export interface ILeaveBalanceAPI {
  data: ILeaveBalanceItemAPI[];
}

export interface ILeaveBalance {
  leaveTypeId: number;
  leaveType: string;
  balance: number;
}

interface IAttendanceItemAPI {
  name: string;
  attendance_date: string;
  start_time: string;
  end_time: string;
}

export interface IAttendanceDetailsAPI {
  data: IAttendanceItemAPI[];
}

export interface IAttendanceItem {
  name: string;
  attendanceDate: string;
  startTime: string;
  endTime: string;
  noOfHours: number;
}

export interface IError {
  ok: boolean;
  message: string;
  status: number;
}
// eslint-disable-next-line @typescript-eslint/no-explicit-any
interface IAsyncState<T = any> {
  error: IError | null;
  loading: boolean;
  value: T;
}

export interface IDefaultRoot {
  userList: IAsyncState<ILabelValue[]>;
  userInfo: IAsyncState<IUserInfo | null>;
  leaveBalance: IAsyncState<ILeaveBalance[]>;
  attendanceDetails: IAsyncState<IAttendanceItem[]>;
  evetDetails: IAsyncState<IEvent[]>;
  allLeaves: IAsyncState<ILeave[]>;
  leaveCountByMonth: IAsyncState<ILeaveCountByMonth[] | []>;
  applyLeave: IAsyncState<IGenericResponse>;
  addAttendance: IAsyncState<IGenericResponse>;
  loginDetails: IAsyncState<ILoginAPI>;
  designationList: IAsyncState<ILabelValue[]>;
  registeration: IAsyncState<IGenericResponse>;
}

export interface IAction {
  type: string;
}

export interface IUserListAction extends IAction {
  payload: IUserAPI[] | null;
}

export interface IUserInfoAction extends IAction {
  payload: IUserInfoDataAPI | null;
}

export interface ILeaveBalanceAction extends IAction {
  payload: ILeaveBalanceItemAPI[] | null;
}

export interface IAttendanceDetailsAction extends IAction {
  payload: IAttendanceItemAPI[] | null;
}

export interface IEventDetailsAction extends IAction {
  payload: IEventItemAPI[] | null;
}

interface ILeaveItemAPI {
  from_date: string;
  to_date: string;
  leave_type: string;
  status: number;
}

export interface IAllLeavesAction extends IAction {
  payload: ILeaveItemAPI[] | null;
}

export interface IAllLeavesAPI {
  data: ILeaveItemAPI[];
}

//TODO
export type TFAttendanceDetailsAction = (
  id: number,
  from: string,
  to: string
) => Promise<Response>;

interface IEventItemAPI {
  attendance_date?: string;
  start_time?: string;
  end_time?: string;
  from_date?: string;
  to_date?: string;
  leave_type?: string;
}

export interface IEventDetailsAPI {
  data: IEventItemAPI[];
}

export const DATE_FORMAT = 'DD-MM-YYYY';
export const DATE_FORMAT_CALENDAR = 'YYYY-MM-DD';

export enum EventType {
  WORKING = 'WORKING',
  LEAVE = 'LEAVE'
}

export interface IEvent {
  title: string;
  start: string;
  end: string;
  allDay?: boolean;
}

export interface IKeyValue {
  [key: string]: number;
}

export interface ILeaveCountByMonth {
  days: number;
  month: string;
}

export interface INewLeave {
  leaveTypeId: number;
  fromDate: string;
  toDate: string;
  userId: number;
}

export interface IGenericResponse<T = unknown> {
  status: number;
  message: string;
  data: T;
}

export interface IGenericAction extends IAction {
  payload: IGenericResponse | null;
}

export const enum NotificationType {
  SUCCESS = 'success',
  ERROR = 'error',
  WARNING = 'warning',
  INFO = 'info'
}

export interface INewAttendance {
  userId: number;
  date: string;
  inTime: string;
  outTime: string;
}

export interface ILoginData {
  userId: number;
  password: string;
}

export interface ILoginAPI {
  token: string;
  issued: number;
  expires: number;
  loginStatus: number;
  message: string;
  userId: number;
}

export interface ILoginAction extends IAction {
  payload: ILoginAPI | null;
}

export interface IRegistration {
  name: string;
  password: string;
  confirmPassword: string;
  email: string;
}

export interface ILabelValue {
  value: number;
  label: string;
}

export interface IDesignationListAPI {
  data: ILabelValue[];
}

export interface IDesignationListAction extends IAction {
  payload: ILabelValue[] | null;
}

export interface IRegisterAPI {
  name: string;
  designation: number;
  manager: number;
  password: string;
  email: string;
}

export interface IRegisterAction extends IAction {
  payload: IRegisterAPI | null;
}

export type ExpirationStatus = 'expired' | 'active' | 'grace';

export interface Session {
  id: number;
  dateCreated: number;
  username: string;
  issued: number;
  expires: number;
}

export type DecodeResult =
  | {
      type: 'valid';
      session: Session;
    }
  | {
      type: 'integrity-error';
    }
  | {
      type: 'invalid-token';
    };

export type TAnyObject = { [key: string]: any }; // eslint-disable-line @typescript-eslint/no-explicit-any

export type TAny = any; // eslint-disable-line @typescript-eslint/no-explicit-any
