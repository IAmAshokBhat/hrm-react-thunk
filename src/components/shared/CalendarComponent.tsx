import Card from '@mui/material/Card';
import { Calendar, EventPropGetter, momentLocalizer } from 'react-big-calendar';
import moment from 'moment';
import 'react-big-calendar/lib/css/react-big-calendar.css';
import { useDispatch, useSelector } from 'react-redux';
import { useEffect } from 'react';
import { fetchEventDetailsAction } from '../../redux/actions';
import { DATE_FORMAT_CALENDAR, EventType, IEvent } from '../../constants';
import { selectLoginDetails, selectMonthlyEvents } from '../../redux/selectors';

export const CalendarComponent = () => {
  const localizer = momentLocalizer(moment);
  const dispatch = useDispatch();
  const startOfMonth = moment().startOf('month').format(DATE_FORMAT_CALENDAR);
  const endOfMonth = moment().endOf('month').format(DATE_FORMAT_CALENDAR);
  const loginDetails = useSelector(selectLoginDetails);

  useEffect(() => {
    dispatch(
      fetchEventDetailsAction(loginDetails.userId, startOfMonth, endOfMonth)
    );
  }, [dispatch, startOfMonth, endOfMonth, loginDetails]);

  const eventList = useSelector(selectMonthlyEvents);

  const eventStyleGenerator: EventPropGetter<IEvent> = (
    event,
    _start,
    _end,
    _isSelected
  ) => ({
    style: {
      background: event.title === EventType.LEAVE ? '#0b9102' : '#2976d2',
      borderRadius: '5px',
      opacity: 0.8,
      color: 'white',
      border: '0px',
      display: 'block'
    }
  });

  return (
    <Card style={{ height: '367px' }}>
      {/* @ts-ignore */}
      <Calendar
        localizer={localizer}
        events={eventList}
        step={60}
        eventPropGetter={eventStyleGenerator}
      />
    </Card>
  );
};
