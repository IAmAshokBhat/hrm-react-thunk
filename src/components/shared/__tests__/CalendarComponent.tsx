// import { screen, waitFor, within } from '@testing-library/react';
import { getEventDetailsAPI } from '../../../api';
import { TAnyObject } from '../../../constants';
import { mockAPICall, renderWithProvider } from '../../../testUtils';
import { CalendarComponent } from '../CalendarComponent';
import { evenDetailsAPI } from './__ignore_tests__/fixtures';
import { defaultState } from '../../__tests__/fixtures';

jest.mock('../../../api', () => ({
  getEventDetailsAPI: jest.fn()
}));

const evenDetailsResponse = {
  ok: true,
  body: evenDetailsAPI
};

describe('Calendar component', () => {
  const generateNode = (initialStoreState: TAnyObject = defaultState) =>
    renderWithProvider(<CalendarComponent />, {
      route: `/dashboard`,
      path: '/dashboard',
      initialStoreState
    });

  beforeEach(() => {
    mockAPICall(getEventDetailsAPI, evenDetailsResponse);
  });

  it('should render and match snapshot', () => {
    const { container } = generateNode();

    expect(container).toMatchSnapshot();
  });
});
