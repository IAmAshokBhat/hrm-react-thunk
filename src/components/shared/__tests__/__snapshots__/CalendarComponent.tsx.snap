// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar component should render and match snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    style="height: 367px;"
  >
    <div
      class="rbc-calendar"
    >
      <div
        class="rbc-toolbar"
      >
        <span
          class="rbc-btn-group"
        >
          <button
            type="button"
          >
            Today
          </button>
          <button
            type="button"
          >
            Back
          </button>
          <button
            type="button"
          >
            Next
          </button>
        </span>
        <span
          class="rbc-toolbar-label"
        >
          February 2022
        </span>
        <span
          class="rbc-btn-group"
        >
          <button
            class="rbc-active"
            type="button"
          >
            Month
          </button>
          <button
            class=""
            type="button"
          >
            Week
          </button>
          <button
            class=""
            type="button"
          >
            Day
          </button>
          <button
            class=""
            type="button"
          >
            Agenda
          </button>
        </span>
      </div>
      <div
        aria-label="Month View"
        class="rbc-month-view"
        role="table"
      >
        <div
          class="rbc-row rbc-month-header"
          role="row"
        >
          <div
            class="rbc-header"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              Sun
            </span>
          </div>
          <div
            class="rbc-header"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              Mon
            </span>
          </div>
          <div
            class="rbc-header"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              Tue
            </span>
          </div>
          <div
            class="rbc-header"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              Wed
            </span>
          </div>
          <div
            class="rbc-header"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              Thu
            </span>
          </div>
          <div
            class="rbc-header"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              Fri
            </span>
          </div>
          <div
            class="rbc-header"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              Sat
            </span>
          </div>
        </div>
        <div
          class="rbc-month-row"
          role="rowgroup"
        >
          <div
            class="rbc-row-bg"
          >
            <div
              class="rbc-day-bg rbc-off-range-bg"
            />
            <div
              class="rbc-day-bg rbc-off-range-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
          </div>
          <div
            class="rbc-row-content"
            role="row"
          >
            <div
              class="rbc-row "
            >
              <div
                class="rbc-date-cell rbc-off-range"
                role="cell"
              >
                <a
                  href="#"
                  role="cell"
                >
                  30
                </a>
              </div>
              <div
                class="rbc-date-cell rbc-off-range"
                role="cell"
              >
                <a
                  href="#"
                  role="cell"
                >
                  31
                </a>
              </div>
              <div
                class="rbc-date-cell"
                role="cell"
              >
                <a
                  href="#"
                  role="cell"
                >
                  01
                </a>
              </div>
              <div
                class="rbc-date-cell"
                role="cell"
              >
                <a
                  href="#"
                  role="cell"
                >
                  02
                </a>
              </div>
              <div
                class="rbc-date-cell"
                role="cell"
              >
                <a
                  href="#"
                  role="cell"
                >
                  03
                </a>
              </div>
              <div
                class="rbc-date-cell"
                role="cell"
              >
                <a
                  href="#"
                  role="cell"
                >
                  04
                </a>
              </div>
              <div
                class="rbc-date-cell"
                role="cell"
              >
                <a
                  href="#"
                  role="cell"
                >
                  05
                </a>
              </div>
            </div>
            <div
              class="rbc-row"
            >
              <div
                class="rbc-row-segment"
                style="flex-basis: 28.57142857142857%; max-width: 28.57142857142857%;"
              >
                 
              </div>
              <div
                class="rbc-row-segment"
                style="flex-basis: 14.285714285714285%; max-width: 14.285714285714285%;"
              >
                <div
                  class="rbc-event rbc-event-allday"
                  style="background: rgb(41, 118, 210); border-radius: 5px; opacity: 0.8; color: white; border: 0px; display: block;"
                  tabindex="0"
                >
                  <div
                    class="rbc-event-content"
                    title="WORKING"
                  >
                    WORKING
                  </div>
                </div>
              </div>
              <div
                class="rbc-row-segment"
                style="flex-basis: 57.14285714285714%; max-width: 57.14285714285714%;"
              >
                <div
                  class="rbc-event rbc-event-allday rbc-event-continues-after"
                  style="background: rgb(11, 145, 2); border-radius: 5px; opacity: 0.8; color: white; border: 0px; display: block;"
                  tabindex="0"
                >
                  <div
                    class="rbc-event-content"
                    title="LEAVE"
                  >
                    LEAVE
                  </div>
                </div>
              </div>
            </div>
            <div
              class="rbc-row"
            >
              <div
                class="rbc-row-segment"
                style="flex-basis: 42.857142857142854%; max-width: 42.857142857142854%;"
              >
                 
              </div>
              <div
                class="rbc-row-segment"
                style="flex-basis: 14.285714285714285%; max-width: 14.285714285714285%;"
              >
                <div
                  class="rbc-event rbc-event-allday"
                  style="background: rgb(41, 118, 210); border-radius: 5px; opacity: 0.8; color: white; border: 0px; display: block;"
                  tabindex="0"
                >
                  <div
                    class="rbc-event-content"
                    title="WORKING"
                  >
                    WORKING
                  </div>
                </div>
              </div>
              <div
                class="rbc-row-segment"
                style="flex-basis: 14.285714285714285%; max-width: 14.285714285714285%;"
              >
                <div
                  class="rbc-event rbc-event-allday"
                  style="background: rgb(41, 118, 210); border-radius: 5px; opacity: 0.8; color: white; border: 0px; display: block;"
                  tabindex="0"
                >
                  <div
                    class="rbc-event-content"
                    title="WORKING"
                  >
                    WORKING
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="rbc-month-row"
          role="rowgroup"
        >
          <div
            class="rbc-row-bg"
          >
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg rbc-today"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
          </div>
          <div
            class="rbc-row-content"
            role="row"
          >
            <div
              class="rbc-row "
            >
              <div
                class="rbc-date-cell"
                role="cell"
              >
                <a
                  href="#"
                  role="cell"
                >
                  06
                </a>
              </div>
              <div
                class="rbc-date-cell"
                role="cell"
              >
                <a
                  href="#"
                  role="cell"
                >
                  07
                </a>
              </div>
              <div
                class="rbc-date-cell"
                role="cell"
              >
                <a
                  href="#"
                  role="cell"
                >
                  08
                </a>
              </div>
              <div
                class="rbc-date-cell"
                role="cell"
              >
                <a
                  href="#"
                  role="cell"
                >
                  09
                </a>
              </div>
              <div
                class="rbc-date-cell rbc-now rbc-current"
                role="cell"
              >
                <a
                  href="#"
                  role="cell"
                >
                  10
                </a>
              </div>
              <div
                class="rbc-date-cell"
                role="cell"
              >
                <a
                  href="#"
                  role="cell"
                >
                  11
                </a>
              </div>
              <div
                class="rbc-date-cell"
                role="cell"
              >
                <a
                  href="#"
                  role="cell"
                >
                  12
                </a>
              </div>
            </div>
            <div
              class="rbc-row"
            >
              <div
                class="rbc-row-segment"
                style="flex-basis: 14.285714285714285%; max-width: 14.285714285714285%;"
              >
                <div
                  class="rbc-event rbc-event-allday rbc-event-continues-prior"
                  style="background: rgb(11, 145, 2); border-radius: 5px; opacity: 0.8; color: white; border: 0px; display: block;"
                  tabindex="0"
                >
                  <div
                    class="rbc-event-content"
                    title="LEAVE"
                  >
                    LEAVE
                  </div>
                </div>
              </div>
              <div
                class="rbc-row-segment"
                style="flex-basis: 28.57142857142857%; max-width: 28.57142857142857%;"
              >
                 
              </div>
              <div
                class="rbc-row-segment"
                style="flex-basis: 14.285714285714285%; max-width: 14.285714285714285%;"
              >
                <div
                  class="rbc-event rbc-event-allday"
                  style="background: rgb(41, 118, 210); border-radius: 5px; opacity: 0.8; color: white; border: 0px; display: block;"
                  tabindex="0"
                >
                  <div
                    class="rbc-event-content"
                    title="WORKING"
                  >
                    WORKING
                  </div>
                </div>
              </div>
              <div
                class="rbc-row-segment"
                style="flex-basis: 28.57142857142857%; max-width: 28.57142857142857%;"
              >
                <div
                  class="rbc-event rbc-event-allday"
                  style="background: rgb(11, 145, 2); border-radius: 5px; opacity: 0.8; color: white; border: 0px; display: block;"
                  tabindex="0"
                >
                  <div
                    class="rbc-event-content"
                    title="LEAVE"
                  >
                    LEAVE
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="rbc-month-row"
          role="rowgroup"
        >
          <div
            class="rbc-row-bg"
          >
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
          </div>
          <div
            class="rbc-row-content"
            role="row"
          >
            <div
              class="rbc-row "
            >
              <div
                class="rbc-date-cell"
                role="cell"
              >
                <a
                  href="#"
                  role="cell"
                >
                  13
                </a>
              </div>
              <div
                class="rbc-date-cell"
                role="cell"
              >
                <a
                  href="#"
                  role="cell"
                >
                  14
                </a>
              </div>
              <div
                class="rbc-date-cell"
                role="cell"
              >
                <a
                  href="#"
                  role="cell"
                >
                  15
                </a>
              </div>
              <div
                class="rbc-date-cell"
                role="cell"
              >
                <a
                  href="#"
                  role="cell"
                >
                  16
                </a>
              </div>
              <div
                class="rbc-date-cell"
                role="cell"
              >
                <a
                  href="#"
                  role="cell"
                >
                  17
                </a>
              </div>
              <div
                class="rbc-date-cell"
                role="cell"
              >
                <a
                  href="#"
                  role="cell"
                >
                  18
                </a>
              </div>
              <div
                class="rbc-date-cell"
                role="cell"
              >
                <a
                  href="#"
                  role="cell"
                >
                  19
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="rbc-month-row"
          role="rowgroup"
        >
          <div
            class="rbc-row-bg"
          >
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
          </div>
          <div
            class="rbc-row-content"
            role="row"
          >
            <div
              class="rbc-row "
            >
              <div
                class="rbc-date-cell"
                role="cell"
              >
                <a
                  href="#"
                  role="cell"
                >
                  20
                </a>
              </div>
              <div
                class="rbc-date-cell"
                role="cell"
              >
                <a
                  href="#"
                  role="cell"
                >
                  21
                </a>
              </div>
              <div
                class="rbc-date-cell"
                role="cell"
              >
                <a
                  href="#"
                  role="cell"
                >
                  22
                </a>
              </div>
              <div
                class="rbc-date-cell"
                role="cell"
              >
                <a
                  href="#"
                  role="cell"
                >
                  23
                </a>
              </div>
              <div
                class="rbc-date-cell"
                role="cell"
              >
                <a
                  href="#"
                  role="cell"
                >
                  24
                </a>
              </div>
              <div
                class="rbc-date-cell"
                role="cell"
              >
                <a
                  href="#"
                  role="cell"
                >
                  25
                </a>
              </div>
              <div
                class="rbc-date-cell"
                role="cell"
              >
                <a
                  href="#"
                  role="cell"
                >
                  26
                </a>
              </div>
            </div>
            <div
              class="rbc-row"
            >
              <div
                class="rbc-row-segment"
                style="flex-basis: 28.57142857142857%; max-width: 28.57142857142857%;"
              >
                 
              </div>
              <div
                class="rbc-row-segment"
                style="flex-basis: 14.285714285714285%; max-width: 14.285714285714285%;"
              >
                <div
                  class="rbc-event rbc-event-allday"
                  style="background: rgb(11, 145, 2); border-radius: 5px; opacity: 0.8; color: white; border: 0px; display: block;"
                  tabindex="0"
                >
                  <div
                    class="rbc-event-content"
                    title="LEAVE"
                  >
                    LEAVE
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="rbc-month-row"
          role="rowgroup"
        >
          <div
            class="rbc-row-bg"
          >
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg rbc-off-range-bg"
            />
            <div
              class="rbc-day-bg rbc-off-range-bg"
            />
            <div
              class="rbc-day-bg rbc-off-range-bg"
            />
            <div
              class="rbc-day-bg rbc-off-range-bg"
            />
            <div
              class="rbc-day-bg rbc-off-range-bg"
            />
          </div>
          <div
            class="rbc-row-content"
            role="row"
          >
            <div
              class="rbc-row "
            >
              <div
                class="rbc-date-cell"
                role="cell"
              >
                <a
                  href="#"
                  role="cell"
                >
                  27
                </a>
              </div>
              <div
                class="rbc-date-cell"
                role="cell"
              >
                <a
                  href="#"
                  role="cell"
                >
                  28
                </a>
              </div>
              <div
                class="rbc-date-cell rbc-off-range"
                role="cell"
              >
                <a
                  href="#"
                  role="cell"
                >
                  01
                </a>
              </div>
              <div
                class="rbc-date-cell rbc-off-range"
                role="cell"
              >
                <a
                  href="#"
                  role="cell"
                >
                  02
                </a>
              </div>
              <div
                class="rbc-date-cell rbc-off-range"
                role="cell"
              >
                <a
                  href="#"
                  role="cell"
                >
                  03
                </a>
              </div>
              <div
                class="rbc-date-cell rbc-off-range"
                role="cell"
              >
                <a
                  href="#"
                  role="cell"
                >
                  04
                </a>
              </div>
              <div
                class="rbc-date-cell rbc-off-range"
                role="cell"
              >
                <a
                  href="#"
                  role="cell"
                >
                  05
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
